uvicorn/supervisors/multiprocess.py

```diff
@@ -42,7 +42,11 @@ def signal_handler(self, sig: signal.Signals, frame: FrameType) -> None:
 
     def run(self) -> None:
         self.startup()
-        self.should_exit.wait()
+
+        while not self.should_exit.wait(self.config.reload_delay):
+            if not any(p.exitcode is None for p in self.processes):
+                break
```

Comment:
> I used the same interval as the one already setup for the reloader, but it could be another one as well.

Yep, I don't think we should use `self.config.reload_delay` here.